avocado/utils/vmimage.py

```diff
@@ -0,0 +1,251 @@
+            if LZMA_CAPABLE:
+                asset_path = self._extract(asset_path)
+            else:
+                raise VMImageError('Cannot uncompress image')
+
+        self.path = self._create_backed_image(asset_path)
+
+    def customize(self, username=None, password=None, install=None,
+                  uninstall=None):
+        """
+        Wrapper around virt-customize to make custom settings in the
+        VM disk image.
+
+        :param username: Username to configure
+        :param password: Password of the provided user
+        :param install: List of packages to install
+        :param uninstall: List of packages to uninstall
+        :type username: string
+        :type password: string
+        :type install: list
+        :type uninstall: list
+        """
+        virt_customize = utils_path.find_command('virt-customize')
+        if not self.path:
+            raise VMImageError('No image available')
+
+        cmd = '%s -a %s' % (virt_customize, self.path)
+
+        if username is not None and password is not None:
+            self.username = username
+            self.password = password
+            cmd += ' --password %s:password:%s' % (username, password)
+
+        if install:
+            cmd += ' --install %s' % ','.join(install)
+
+        if uninstall:
+            cmd += ' --uninstall %s' % ','.join(uninstall)
+
+        process.run(cmd)
+
+    def remove(self):
+        """
+        Removes the VM Image
+        """
+        try:
+            os.remove(self.path)
+        except:
+            pass
+        self.path = None
+
+    @staticmethod
+    def _create_backed_image(source_image):
+        """
+        Creates a Qcow2 backed image using the original image
+        as backing file.
+        """
+        qemu_img = utils_path.find_command('qemu-img')
+        new_image = '%s-%s' % (source_image, uuid.uuid4())
+        cmd = '%s create -f qcow2 -b %s %s' % (qemu_img,
+                                               source_image,
+                                               new_image)
+        process.run(cmd)
+        return new_image
+
+    @staticmethod
+    def _extract(path, force=False):
+        """
+        Extracts a XZ compressed file to the same directory.
+        """
+        extracted_file = os.path.splitext(path)[0]
+        if not force and os.path.exists(extracted_file):
+            return extracted_file
+        with open(path, 'r') as file_obj:
+            with open(extracted_file, 'wb') as newfile_obj:
+                newfile_obj.write(lzma.decompress(file_obj.read()))
+        return extracted_file
+
+
+class CentOS(Image):
+    def __init__(self, name=None, major=None, minor=None, build=None,
+                 arch=None, checksum=None, algorithm=None, cache_dir=None):
+        """
+        CentOS Qcow2 GenericCloud compressed image
+        """
+        required_args = ['major', 'build', 'arch']
+        for arg in required_args:
+            if eval(arg) is None:
```

Comment:
Seriously, using eval just to check the presense? Please just use `None in [major, build, arch]` instead... (or at least locals, but definitely not eval)

Reply:
Yes.. not sure how it ended up there :)
Anyway, there's a new version already.